src/pyqasm/visitor.py

```diff
@@ -89,7 +89,9 @@ def __init__(
         self._curr_scope: int = 0
         self._label_scope_level: dict[int, set] = {self._curr_scope: set()}
         self._recording_ext_gate_depth = False
-
+        self._is_branching_statement: bool = False
```

Comment:
```suggestion
        self._in_branching_statement: bool = False
```

Let's rename this as we want to signify that we are inside a branching statement